src/modm/driver/pwm/lp503x_impl.hpp.in

```diff
@@ -0,0 +1,184 @@
+	if(channel >= Channels) {
+		RF_RETURN(false);
+	}
+
+	RF_END_RETURN_CALL(writeRegister(reg, value));
+}
+
+%% if options.multichannel_write
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setChannelBrightnessValues(uint8_t startChannel,
+														const uint8_t* values,
+														uint8_t numChannels)
+{
+	RF_BEGIN();
+	if(numChannels == 0 || (startChannel + numChannels) > Channels) {
+		RF_RETURN(false);
+	}
+
+	RF_END_RETURN_CALL(writeRegisters(
+		static_cast<Register>(uint8_t(Register::Out0Color) + startChannel),
+		values, numChannels));
+}
+%% endif
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setLedGroupBrightness(uint8_t index, uint8_t value)
+{
+	const Register reg = static_cast<Register>(uint8_t(Register::Led0Brightness) + index);
+
+	RF_BEGIN();
+	if(index >= (Channels / 3)) {
+		RF_RETURN(false);
+	}
+
+	RF_END_RETURN_CALL(writeRegister(reg, value));
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setBankModeEnabled(LedBankMode_t bankMode)
+{
+	RF_BEGIN();
+	buffer[0] = static_cast<uint8_t>(Register::LedConfig0);
+	buffer[1] = bankMode.value & 0xff;
+	buffer[2] = (bankMode.value & 0xf00) >> 8;
+	this->transaction.configureWrite(buffer, 3);
+	RF_END_RETURN_CALL(this->runTransaction());
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setBankABrightness(uint8_t value)
+{
+	return writeRegister(Register::BankAColor, value);
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setBankBBrightness(uint8_t value)
+{
+	return writeRegister(Register::BankBColor, value);
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setBankCBrightness(uint8_t value)
+{
+	return writeRegister(Register::BankCColor, value);
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::setBankBrightness(uint8_t value)
+{
+	return writeRegister(Register::BankBrightness, value);
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::writeRegister(Register reg, RegisterValue_t value)
+{
+	return writeRegister(reg, value.value);
+}
+
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::writeRegister(Register reg, uint8_t value)
+{
+	RF_BEGIN();
+	buffer[0] = static_cast<uint8_t>(reg);
+	buffer[1] = value;
+	this->transaction.configureWrite(buffer, 2);
+	RF_END_RETURN_CALL(this->runTransaction());
+}
+
+%% if options.multichannel_write
+template<typename I2cMaster, size_t Channels>
+ResumableResult<bool>
+Lp503x<I2cMaster, Channels>::writeRegisters(Register startRegister,
+										   const uint8_t* values, uint8_t numValues)
+{
+	RF_BEGIN();
+
+	if((numValues + 1u) > sizeof(buffer)) {
```

Comment:
Why not 
```suggestion
	if(numValues > (sizeof(buffer) - 1)) {
```

Reply:
I think this is nicer, since numValues can overflow (when passing 255) and thus the check fails. `sizeof(buffer)` is always at least > 1.

Reply:
I can change it if you like, but there is no overflow with 255. The `uint8_t` value is promoted to `unsigned int`. When you add two `uint8_t` you always get a result of unsigned int.
`decltype(' ' + ' ')` is actually int or unsigned int dependent on the signedness of char on your platform.

Reply:
🤯🤯🤯🤯🤯🤯 THIS LANGUAGE IS COMPLETELY BONKERS!!! 🤯🤯🤯🤯🤯🤯

(ok, fine)

Reply:
Eh, adding `uint8_t` + `uint8_t` gives int not unsigned int, these C int promotion rules ...
But `numValues + 1u` is indeed unsigned int.